### PR TITLE
Add net_ rpc methods

### PR DESCRIPTION
**Pull request by LukasGasior1**
wip: needs tests

### CODE REVIEW

File: src/main/resources/application.conf [hunk 1]
```diff
@@ -14,6 +14,8 @@ etc-client {
 
   network {
 
+    protocol-version = "1"
```
Comment (adamsmo) [inline src/main/resources/application.conf]: this is json RPC version ?

Comment (AlanVerbner) [inline src/main/resources/application.conf]: [Here](https://github.com/ethereum/wiki/wiki/JSON-RPC#net_version) it says: 

> Returns the current network protocol version.

Parity is returning me `1` but take a look at this: https://github.com/ethereum/EIPs/issues/611 :thinking: 

> go-ethereum and Parity too currently returns the eth network ID, which is probably the worse possible interpretation of this field based on its name and description.

File: src/main/scala/io/iohk/ethereum/jsonrpc/NetService.scala [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    }
+  }
+
+  def peerCount(req: PeerCountRequest): Future[PeerCountResponse] = {
+    import akka.pattern.ask
+    implicit val timeout = Timeout(2.seconds)
+
+    (peerManager ? PeerManagerActor.GetPeers)
+      .mapTo[PeerManagerActor.Peers]
+      .map { peers => PeerCountResponse(peers.handshaked.size) }
```
Comment (AlanVerbner) [inline src/main/scala/io/iohk/ethereum/jsonrpc/NetService.scala]: I haven't checked other implementations yet, but as descriptions are not always the best...have you considered blacklisted peers? Should we count them on this query?

Comment (LukasGasior1) [inline src/main/scala/io/iohk/ethereum/jsonrpc/NetService.scala]: I guess it's up to us what we return here :) Blacklisted peers are currently included - it's easier and also blacklisted peers are really just for usage in sync, they are still valid handshaked peers.

File: src/main/resources/application.conf (hunk 1 above)
Comment (LukasGasior1) [inline src/main/resources/application.conf]: Seems, there's a chance of deprecating this method in near future

File: src/test/scala/io/iohk/ethereum/jsonrpc/NetServiceSpec.scala [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+import org.scalamock.scalatest.MockFactory
+import org.scalatest.{FlatSpec, Matchers}
+
+import scala.concurrent.Await
+import scala.concurrent.duration._
+import scala.concurrent.ExecutionContext.Implicits.global
+
+class NetServiceSpec extends FlatSpec with Matchers with MockFactory {
+
+  "NetService" should "return handshaked peer count" in new TestSetup {
```
Comment (rtkaczyk) [inline src/test/scala/io/iohk/ethereum/jsonrpc/NetServiceSpec.scala]: I realise `version` is trivial but I would still have it tested here, and in `JsonRpcControllerSpec`

File: src/main/scala/io/iohk/ethereum/jsonrpc/JsonMethodsImplicits.scala [hunk 4]
```diff
@@ -29,10 +30,24 @@ object JsonMethodsImplicits {
...
+  }
+
+  implicit val net_listening = new JsonDecoder[ListeningRequest] with JsonEncoder[ListeningResponse] {
+    override def decodeJson(params: Option[JArray]): Either[JsonRpcError, ListeningRequest] = Right(ListeningRequest())
+    override def encodeJson(t: ListeningResponse): JValue = t.value
+  }
+
+  implicit val net_peerCount = new JsonDecoder[PeerCountRequest] with JsonEncoder[PeerCountResponse] {
+    override def decodeJson(params: Option[JArray]): Either[JsonRpcError, PeerCountRequest] = Right(PeerCountRequest())
+    override def encodeJson(t: PeerCountResponse): JValue = encodeAsHex(t.value)
```
Comment (ntallar) [inline src/main/scala/io/iohk/ethereum/jsonrpc/JsonMethodsImplicits.scala]: Maybe we should also remove the leading zeros here as the returned value is a `Quantity`?

**Comment by ntallar** [comment]
Apart from my comment on the leading zeroes, LGTM!

File: src/main/scala/io/iohk/ethereum/jsonrpc/JsonMethodsImplicits.scala (hunk 4 above)
Comment (LukasGasior1) [inline src/main/scala/io/iohk/ethereum/jsonrpc/JsonMethodsImplicits.scala]: nice catch, done

File: src/main/scala/io/iohk/ethereum/jsonrpc/JsonMethodsImplicits.scala [hunk 5]
```diff
@@ -43,7 +58,8 @@ object JsonMethodsImplicits {
     JString(s"0x${Hex.toHexString(input.toArray[Byte])}")
 
   private def encodeAsHex(input: BigInt): JString =
-    JString(s"0x${Hex.toHexString(input.toByteArray)}")
+    if (input == 0) JString("0x0")
+    else JString(s"0x${Hex.toHexString(input.toByteArray).dropWhile(_ == '0')}")
```
Comment (rtkaczyk) [inline src/main/scala/io/iohk/ethereum/jsonrpc/JsonMethodsImplicits.scala]: How about `0x + input.toString(16)` ?

Comment (LukasGasior1) [inline src/main/scala/io/iohk/ethereum/jsonrpc/JsonMethodsImplicits.scala]: done

File: src/main/scala/io/iohk/ethereum/jsonrpc/JsonMethodsImplicits.scala [hunk 6]
```diff
@@ -43,7 +58,8 @@ object JsonMethodsImplicits {
     JString(s"0x${Hex.toHexString(input.toArray[Byte])}")
 
   private def encodeAsHex(input: BigInt): JString =
-    JString(s"0x${Hex.toHexString(input.toByteArray)}")
+    if (input == 0) JString("0x0")
+    else JString(s"0x${input.toString(16)}")
```
Comment (rtkaczyk) [inline src/main/scala/io/iohk/ethereum/jsonrpc/JsonMethodsImplicits.scala]: I meant instead of the `if` 😉

Comment (LukasGasior1) [inline src/main/scala/io/iohk/ethereum/jsonrpc/JsonMethodsImplicits.scala]: right... :D

**Pull request merged**